bitseq/sequence.go

```diff
@@ -186,12 +186,23 @@ func (h *Handle) getCopy() *Handle {
 	}
 }
 
+// SetAnyInRange atomically sets the first unset bit in the specified range in the sequence and returns the corresponding ordinal
+func (h *Handle) SetAnyInRange(start, end uint32) (uint32, error) {
+	if end-start <= 0 || end > h.Bits() {
+		return invalidPos, fmt.Errorf("invalid bit range [%d, %d]", start, end)
+	}
+	if h.Unselected() == 0 {
+		return invalidPos, errNoBitAvailable
+	}
+	return h.set(0, start, end, true, false)
+}
+
 // SetAny atomically sets the first unset bit in the sequence and returns the corresponding ordinal
 func (h *Handle) SetAny() (uint32, error) {
 	if h.Unselected() == 0 {
 		return invalidPos, errNoBitAvailable
 	}
-	return h.set(0, true, false)
+	return h.set(0, 0, 0, true, false)
```

Comment:
Can't you pass `h.Bits()` for `end` value so that you can avoid checking `end != 0` down below?